src/QuadForm/Herm/Genus.jl

```diff
@@ -479,7 +479,13 @@ Given a local genus, return a Hermitian lattice contained in this genus.
 """
 function representative(G::LocalGenusHerm)
   E = G.E
-  return lattice(hermitian_space(E, gram_matrix(G)))
+  p = prime(G)
+  L = lattice(hermitian_space(E, gram_matrix(G)))
+  S = ideal_type(base_ring(base_ring(L)))
+  symbols = Dict{S, LocalGenusHerm{typeof(base_field(L)),S}}()
+  set_attribute!(L, :local_genera => symbols)
+  symbols[p] = G
+  return L
```

Comment:
This does not make sense to me. 
What is it supposed to fix? `representative(g) in g` works for me. Can you give an example where it fails?

Reply:
The lattice `L = representative(g)` will remember its local genus at `p`. `genus(L, p)` should then be instant.

Reply:
```
julia> K,a = quadratic_field(2)
(Real quadratic field defined by x^2 - 2, sqrt(2))

julia> Kt, t = K["t"]
(Univariate Polynomial Ring in t over K, t)

julia> OK = maximal_order(K)
Maximal order of Real quadratic field defined by x^2 - 2 
with basis nf_elem[1, sqrt(2)]

julia> E,b = number_field(t^2-a, "b")
(Relative number field over with defining polynomial t^2 - sqrt(2)
 over Real quadratic field defined by x^2 - 2, b)

julia> p = prime_ideals_over(OK,5)[1]
<5, 5>
Norm: 25
Minimum: 5
principal generator 5
two normal wrt: 5

julia> g = genus(HermLat, E, p, [(0,1,1,0),(2,2,-1,-1)])
Local genus symbol (scale, rank, det) at <5, 5>:
(0, 1, +)(2, 2, -)

julia> L = representative(g)
Hermitian lattice of rank 3 and degree 3
over
Relative maximal order with pseudo-basis (1) * 1//1 * <1, 1>, (b) * 1//1 * <1, 1>

julia> L in g
false
```
The problem is when we do the check `L in g`, if `get_attribute(L, :local_genera)` doesn't have a key `p`, it will recreate the local genus `g` but as julia object, this new version of `g` is distinct from the one we have started.

Reply:
For `HermLat` the way the representative is constructed, it doesn't keep systematically in memory that `genus(L,p) == g`... 
```
julia> genus(L,p) == g
false
```

Reply:
Is the equality of local genera missing?

Reply:
A priori no, at least the method `function ==(G1::LocalGenusHerm, G2::LocalGenusHerm)` exists in /src/QuadForm/Herm/Genus.jl from the line 698

Reply:
Something does not look right here. In particular
```
julia> L in g
false
```

Reply:
This is coming mainly from the function `genus(L,p)` which returns the value at the key `p` of `get_attribute(L, :local_genera)` and if it doesn't exist, it calls `Hecke._genus(L,p)` which creates a new copy of `g`, but which is not the same as `g`

Reply:
It should not matter, since we define == for local genera. It does not fall back to === (which tests if the objects are identical).

Reply:
That might help 
```
julia> det(genus(L,p))
1

julia> det(g)
-1
```

Reply:
Yes. There is a bug somewhere.

Reply:
In this case `p` is split in E, so the genus symbol can never have a `-` as the determinant class? It seems
```
g = genus(HermLat, E, p, [(0,1,1,0),(2,2,-1,-1)])
```
makes no sense (this is the same as `[(0,1,1),(2,2,-1)])` Can you confirm @simonbrandhorst?

Reply:
Nevermind, I am not sure what I wrote makes sense. I think I found the problem.

Reply:
Still confused. I am looking at Prop. 3.3.5 of http://www.math.rwth-aachen.de/~Markus.Kirschmer/papers/herm.pdf. Consider `(2, 2 -1)`, so rank 2 of scale 2. We are not ramified, so in case 2 and the lattice looks like <z^2, z^2> where z is a uniformizer of p (notation as in Stevells example). So the determinant is z^4, which is a local norm:
```
julia> Hecke.islocal_norm(E, elem_in_nf(uniformizer(p)^4), p)
true
```
So I don't think (2, 2, -1) is a genus symbol?

P.S.: In the unramified case, scale and rank always determine the detclass.

Reply:
@StevellM Where did you find that genus symbol?